UPGRADE-14.0.md

```diff
@@ -764,10 +764,83 @@ Follow the instructions in relevant sections, e.g. `shopsys/coding-standards` or
+-   fix broken drag and drop in GrapesJS in Safari ([#2966](https://github.com/shopsys/shopsys/pull/2966))
     -   see #project-base-diff to update your project
 -   add Category ID to CategoryHierarchyItem ([#2962](https://github.com/shopsys/shopsys/pull/2962))
     -   see #project-base-diff to update your project
+-   add hreflang feature ([#2970](https://github.com/shopsys/shopsys/pull/2970))
+    -   method `Shopsys\FrameworkBundle\Model\Blog\Article\Elasticsearch\BlogArticleExportQueueFacade::__construct()` changed its interface:
+    ```diff
+        public function __construct(
+            Redis $redisQueue,
+    +       protected readonly BlogArticleFacade $blogArticleFacade,
+    ```
+    -   method `Shopsys\FrameworkBundle\Model\Blog\Article\Elasticsearch\BlogArticleExportRepository::__construct()` changed its interface:
+    ```diff
+        public function __construct(
+            protected readonly EntityManagerInterface $em,
+            protected readonly BlogArticleRepository $blogArticleRepository,
+            protected readonly FriendlyUrlFacade $friendlyUrlFacade,
+            protected readonly BreadcrumbFacade $breadcrumbFacade,
+            protected readonly GrapesJsParser $grapesJsParser,
+            protected readonly ImageFacade $imageFacade,
+            protected readonly Domain $domain,
+    +       protected readonly HreflangLinksFacade $hreflangLinksFacade,
+    ```
+    -   method `Shopsys\FrameworkBundle\Model\Product\Elasticsearch\ProductExportRepository::__construct()` changed its interface:
+    ```diff
+        public function __construct(
+            protected readonly EntityManagerInterface $em,
+            protected readonly ParameterRepository $parameterRepository,
+            protected readonly ProductFacade $productFacade,
+            protected readonly FriendlyUrlRepository $friendlyUrlRepository,
+            protected readonly ProductVisibilityFacade $productVisibilityFacade,
+            protected readonly FriendlyUrlFacade $friendlyUrlFacade,
+            protected readonly CategoryFacade $categoryFacade,
+            protected readonly ProductAccessoryFacade $productAccessoryFacade,
+            protected readonly BrandCachedFacade $brandCachedFacade,
+            protected readonly ProductAvailabilityFacade $productAvailabilityFacade,
+    +       protected readonly HreflangLinksFacade $hreflangLinksFacade,
+    ```
+    -   method `Shopsys\FrameworkBundle\Model\Seo\SeoSettingFacade::__construct()` changed its interface:
+    ```diff
+        public function __construct(
+            protected readonly Setting $setting,
+    +       protected readonly ProductRecalculationDispatcher $productRecalculationDispatcher,
+    +       protected readonly BlogArticleExportQueueFacade $blogArticleExportQueueFacade,
+    ```
+    -   method `Shopsys\FrontendApiBundle\Model\Blog\Category\BlogCategoryResolverMap::__construct()` changed its interface:
```

Comment:
`BlogCategoryResolverMap` was added in this version, so no upgrade notes are needed here, am I right?

Reply:
Right! removed